src/runtime/HalideBuffer.h

```diff
@@ -1674,6 +1682,14 @@ class Buffer {
         return dst;
     }
 
+    /** Make a buffer with the same shape and type and memory nesting order as
+     * this buffer; this is syntactic sugar for the static make_with_shape_of method
+     * when src and dst types are identical. */
+    Buffer<T, D> make_with_shape_of(void *(*allocate_fn)(size_t) = nullptr,
```

Comment:
Not sure about this one. buf.make is a bit ambiguous IMHO, since it could be either returning a new buffer (it is) or replacing the contents of the buffer it is called on.

I'm not sure about this suggestion either... but a standalone (non-member) function would avoid that problem, and get template argument deduction too.

Reply:
The goal here is that code of the form

```
Buffer<float> input = ...
Buffer<float> output = Buffer<float>::make_with_shape_of(input);
```
is stuttering and repeats info, and can be made simpler with
```
Buffer<float> input = ...
Buffer<float> output = input::make_with_shape_of();
```
But yes, the existing `make_` methods are all static, so maybe a better name is called for.

Reply:
(BTW: I'm fine to leave this one out of this PR if we can't come to quick agreement on naming; the `copy` method improvement is the only part of this that I will actually get blocked on soon.)